auditbeat/module/file_integrity/metricset.go

```diff
@@ -167,6 +167,13 @@ func (ms *MetricSet) init(reporter mb.PushReporterV2) bool {
 }
 
 func (ms *MetricSet) reportEvent(reporter mb.PushReporterV2, event *Event) bool {
+	// ignore excluded paths
+	for _, matcher := range ms.config.ExcludeMatchers {
```

Comment:
While it is most convenient to apply the filtering here, this wouldn't be much different that using a drop_event processor with a regexp condition. In my mind the benefit of having a dedicated `exclude_files` option is that the filter can be applied earlier in order to save the CPU and I/O cycles associated with reading the file and hashing it. Would it be feasible to apply this filter earlier?

It would need to be applied to the scanner and fsnotify/fsevents readers. I think the events that are triggered by `purgeDeleted` would need filtered as well.